daemon/pod/provision.go

```diff
@@ -471,3 +495,30 @@ func (p *XPod) sandboxShareDir() string {
 	}
 	return filepath.Join(hypervisor.BaseDir, p.sandbox.Id, hypervisor.ShareDirTag)
 }
+
+func (p *XPod) resetInitNotification() {
+	select {
+	case <-p.initChan:
+		p.Log(TRACE, "drain the init notification chan")
+	default:
+		p.Log(TRACE, "init notification chan is empty")
+	}
+}
+
+func (p *XPod) waitPodRun(activity string) (bool, error) {
+	if p.IsRunning() {
+		p.Log(TRACE, "pod is running, proceed %s", activity)
+		return true, nil
+	}
+	if !p.IsAlive() {
+		p.Log(ERROR, "pod is not alive, cannot %s", activity)
+		return false, errors.ErrPodNotAlive.WithArgs(p.Id())
+	}
+	res, ok := <-p.initChan
+	if !ok {
+		p.Log(ERROR, "init chan broken")
+		return false, nil
+	}
+	p.initChan <- res
```

Comment:
I don't think so. But I am thinking whether I should change to Cond/Wait here....